gita/__main__.py

```diff
@@ -83,7 +83,12 @@ def f_info(args: argparse.Namespace):
 
 def f_clone(args: argparse.Namespace):
     path = Path.cwd()
-    errors = utils.exec_async_tasks(
+    if args.preserve_path:
+        utils.exec_async_tasks(
+            utils.run_async(repo_name, path, ['git', 'clone', url, abs_path])
```

Comment:
What happens when this fails? For example, the user don't have write access to that folder?

Reply:
I just gave it a try:

I created _/home/user/parent_ and removed user's permissions.

Trying to clone a repo _child_ led to:
> child: fatal: could not create leading directories of '/home/user/parent/subdir/child': Permission denied

Trying to clone a repo _parent_ led to:
> parent: fatal: destination path '/home/user/parent' already exists and is not an empty directory.

The second error is a bit misleading because it's not really empty, but it's the error git provides.

Reply:
Thanks.  For the second case, that's the expected error even if you have write access. Copying into an existing non-empty directory would make the repo dirty and possibly overwriting pre-existing files, thus is disallowed.